.github/workflows/ci.yaml

```diff
@@ -36,12 +36,31 @@ jobs:
         run: make build-examples
   build-test:
     runs-on: ubuntu-latest
+    services:
```

Comment:
Given we now have testcontainers-go for that, I'd encourage to use the existing module for pgvector. I offer myself to assist in it. It will improve the dev experience, as we'll be able to run the very same tests the CI runs, but locally.

Reply:
Done. pls take a look again.

Reply:
Thanks! In only 15s the code runs 10 containers! 👏👏👏

As a follow up, one container could be used, and each test would use it's own snapshot: https://golang.testcontainers.org/modules/postgres/#using-snapshots